modules/generator/processor/servicegraphs/store/edge.go

```diff
@@ -37,12 +37,22 @@ type Edge struct {
 	SpanMultiplier float64
 }
 
-func newEdge(key string, ttl time.Duration) *Edge {
-	return &Edge{
-		key:        key,
-		Dimensions: make(map[string]string),
-		expiration: time.Now().Add(ttl).Unix(),
+// zeroStateEdge resets the Edge to its zero state.
+// Useful for reusing an Edge without allocating a new one.
+func zeroStateEdge(e *Edge) {
+	e.TraceID = ""
+	e.ConnectionType = Unknown
+	e.ServerService = ""
+	e.ClientService = ""
+	e.ServerLatencySec = 0
+	e.ClientLatencySec = 0
+	e.Failed = false
+	for k := range e.Dimensions {
+		// saves 30ns/op, 50 B/op, 1 allocs/op
```

Comment:
use the `clear(<map>)` func?

Reply:
can do - similar benchmarks now, maybe also better at volume anyway

Reply:
probably no perf change, but i prefer it for clarity. 

also maybe some future perf improvements would make it into `clear()` that wouldn't be in the loop?